libs/paymentProcessor.js

```diff
@@ -182,81 +190,49 @@ function SetupForPool(logger, poolOptions, setupFinished){
 
             /* Does a batch rpc call to daemon with all the transaction hashes to see if they are confirmed yet.
                It also adds the block reward amount to the round object - which the daemon gives also gives us. */
-            function(workers, rounds, callback){
+            function (workers, rounds, callback) {
 
-                var batchRPCcommand = rounds.map(function(r){
-                    return ['gettransaction', [r.txHash]];
+                let batchRPCcommand = rounds.map(function (r) {
+                    return ['eth_getBlockByNumber', [r.height, false]];
                 });
 
-                batchRPCcommand.push(['getaccount', [poolOptions.address]]);
+                //batchRPCcommand.push(['getaccount', [poolOptions.address]]);
 
                 startRPCTimer();
-                daemon.batchCmd(batchRPCcommand, function(error, txDetails){
+                daemon.batchCmd(batchRPCcommand, function (error, blocksDetails) {
                     endRPCTimer();
 
-                    if (error || !txDetails){
-                        logger.error(logSystem, logComponent, 'Check finished - daemon rpc error with batch gettransactions '
+                    if (error || !blocksDetails) {
+                        logger.error(logSystem, logComponent, 'Check finished - daemon rpc error with batch getBlockByHash '
                             + JSON.stringify(error));
                         callback(true);
                         return;
                     }
 
-                    var addressAccount;
-
-                    txDetails.forEach(function(tx, i){
+                    let addressAccount = poolOptions.address;
 
-                        if (i === txDetails.length - 1){
-                            addressAccount = tx.result;
-                            return;
-                        }
+                    blocksDetails.forEach(function (block, i) {
 
-                        var round = rounds[i];
-
-                        if (tx.error && tx.error.code === -5){
-                            logger.warning(logSystem, logComponent, 'Daemon reports invalid transaction: ' + round.txHash);
+                        let round = rounds[i];
+                        //we have invalid block details returned, either from invalid hash or block has been mined
+                        if (!block.result.miner) {
```

Comment:
do we get blocks without a miner?

Reply:
Just saw that my comment is partly wrong. Should read "block has not been mined". Will modify